app/Http/Controllers/FreeTrailController.php

```diff
@@ -0,0 +1,258 @@
+
+        $this->invoice = new Invoice();
+
+        $this->invoiceItem = new InvoiceItem();
+
+        $this->order = new Order();
+
+        $this->subscription = new Subscription();
+    }
+
+    /**
+     * Get the auth user id
+     * Check the first_time_login is not equal to zero in DB to correspaonding,if its not change into one.
+     *
+     * @return order
+     */
+    public function firstLoginAtem(Request $request)
+    {
+        try {
+            if (! Auth::check()) {
+                return redirect('login')->back()->with('fails', \Lang::get('message.free-login'));
+            }
+
+            $userId = $request->get('id');
+            if (Auth::user()->id == $userId) {
+                $userLogin = User::find($userId);
+                if ($userLogin->first_time_login != 0) {
+                    return errorResponse(Lang::get('message.false'), 400);
+                }
+                User::where('id', $userId)->update(['first_time_login' => 1]);
+
+                $this->generateFreetrailInvoice();
+
+                $this->createFreetrailInvoiceItems();
+
+                $this->executeFreetrailOrder();
+
+                return successResponse(Lang::get('message.succs_fre'), $this->executeFreetrailOrder(), 200);
+            }
+        } catch (\Exception $ex) {
+            app('log')->error($ex->getMessage());
+
+            throw new \Exception('Can not Generate Subscription');
+        }
+    }
+
+    /**
+     * Generate invoice from client panel for free trial.
+     *
+     * @throws \Exception
+     */
+    private function generateFreetrailInvoice()
+    {
+        try {
+            $tax_rule = new \App\Model\Payment\TaxOption();
+            $rule = $tax_rule->findOrFail(1);
+            $rounding = $rule->rounding;
+            $user_id = \Auth::user()->id;
+            $grand_total = $rounding ? round(\Cart::getTotal()) : \Cart::getTotal();
+            $number = rand(11111111, 99999999);
+            $date = \Carbon\Carbon::now();
+            $currency = \Session::has('cart_currency') ? \Session::get('cart_currency') : getCurrencyForClient(\Auth::user()->country);
+            $invoice = $this->invoice->create(['user_id' => $user_id, 'number' => $number, 'date'=> $date, 'grand_total' => $grand_total, 'status' => 'pending',
+                'currency' => $currency, ]);
+
+            return $invoice;
+        } catch (\Exception $ex) {
+            app('log')->error($ex->getMessage());
+
+            throw new \Exception('Can not Generate Subscription');
+        }
+    }
+
+    /**
+     * Generate invoice items for free trial.
+     *
+     * @throws \Exception
+     */
+    private function createFreetrailInvoiceItems()
+    {
+        try {
+            $cart = \Cart::getContent();
+            $userId = \Auth::user()->id;
+            $invoice = $this->invoice->where('user_id', $userId)->first();
+            $invoiceid = $invoice->id;
+            $invoiceItem = $this->invoiceItem->create([
```

Comment:
try getting these values dynamically, rather than hardcoding them. Because when we go live with this it'll require changes. So it's better if we think of something now

Reply:
done